### PR TITLE
User context includes data from database

### LINKED ISSUE #221: User context from DB

### PR DESCRIPTION
**Pull request by tiago86g**
# Description
Now userContext includes data from database
Once we are taking user data from database instead from Firebase I need to overwrite part of #213 which was taking data from Firebase.

Fixes #221 

# How to test?
run "npm run dev" and log user from userContext

# Checklist
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

File: src/client/App.js [hunk 1]
```diff
@@ -11,13 +11,32 @@ import Header from './components/NavigationHeader/NavigationHeader';
...
+
+  useEffect(() => {
+    const fetchUserFromDatabase = async () => {
+      try {
+        const { uid } = await user;
+        const url = `/api/users/current/?uid=`;
+        const userFromDB = await fetchWithAuth(url + uid);
+        return setCurrentUser(userFromDB);
+      } catch (err) {
+        return <p>{err}</p>;
```
Comment (tpetrina) [inline src/client/App.js:29]: You cannot return this from `useEffect`. The only thing you can return is a cleanup function.
If you need to show an error, use another state variable called `err` with `useState`.

File: src/client/App.js [hunk 2]
```diff
@@ -11,13 +11,32 @@ import Header from './components/NavigationHeader/NavigationHeader';
...
 function App() {
+  const [currentUser, setCurrentUser] = useState();
   const { isAuthenticated, isLoading, user } = useAuthentication();
+
+  useEffect(() => {
+    const fetchUserFromDatabase = async () => {
+      try {
+        const { uid } = await user;
+        const url = `/api/users/current/?uid=`;
+        const userFromDB = await fetchWithAuth(url + uid);
```
Comment (tpetrina) [inline src/client/App.js:26]: Prefer string interpolation `${}` to string concantenation "" + "".

File: src/client/App.js [hunk 3]
```diff
@@ -11,13 +11,32 @@ import Header from './components/NavigationHeader/NavigationHeader';
...
+        return setCurrentUser(userFromDB);
```
Comment (tpetrina) [inline src/client/App.js:27]: You probably want to combine the data. Use spreading e.g.:
```
setCurrentUser({
  ...currentUser,
  ...userFromDB
});```

File: src/server/api/controllers/users-controller.js [hunk 4]
```diff
@@ -12,7 +12,7 @@ const getUsers = async () => {
 
 const getUserByUid = async (uid) => {
   const user = await knex('users')
-    .select('user_name', 'id')
+    .select('*')
```
Comment (tpetrina) [inline src/server/api/controllers/users-controller.js:15]: I would suggest not returning everything (like `created_at`, etc).
Another suggestion is to rename the variables to something more natural e.g. `user_name` to `userName`, `profile_image` to `profileImage`.

Comment (tiago86g) [reply src/server/api/controllers/users-controller.js:15]: Yes, I was thinking about it, the database naming is all snake_case and we need to use it in be camelCase

Comment (tiago86g) [reply src/server/api/controllers/users-controller.js:15]: Should I rename it here or in the front-end?

File: src/client/App.js (hunk 3 above)
Comment (tiago86g) [reply src/client/App.js:27]: In that case userFromDB I want just replace it because userFromDB has everything.

File: src/client/App.js [hunk 5]
```diff
@@ -11,23 +11,36 @@ import Header from './components/NavigationHeader/NavigationHeader';
...
-      : {
-          uid: user.uid,
-          email: user.email,
-          displayName: user.displayName,
-        };
+
+  useEffect(() => {
+    const fetchUserFromDatabase = async () => {
+      try {
+        const { uid } = await user;
```
Comment (tpetrina) [inline src/client/App.js:25]: Wait, can you `await` a simple variable?

**Review by tpetrina — APPROVED** [review]

File: src/client/App.js [hunk 6]
```diff
@@ -11,23 +11,36 @@ import Header from './components/NavigationHeader/NavigationHeader';
...
+        const url = `/api/users/current/?uid=`;
+        await fetchWithAuth(`${url}${uid}`).then((userData) =>
```
Comment (benna100) [inline src/client/App.js:27]: You are using a mix of await and promise. Stick to one

**Review by benna100 — CHANGES_REQUESTED** [review]

**Review by tpetrina — APPROVED** [review]

File: src/server/api/controllers/users-controller.js (hunk 4 above)
Comment (tpetrina) [reply src/server/api/controllers/users-controller.js]: Controllers should take snake case and convert to camelCase. That is why we have this entire layer - to prepare data in a nice way for our frontend consumers.

**Pull request merged**